runtime/codert_vm/cnathelp.cpp

```diff
@@ -885,13 +885,69 @@ old_fast_jitPutFlattenableStaticField(J9VMThread *currentThread)
 void* J9FASTCALL
 old_slow_jitLoadFlattenableArrayElement(J9VMThread *currentThread)
 {
-	return NULL;
+	SLOW_JIT_HELPER_PROLOGUE();
+	j9object_t arrayObject = (j9object_t)currentThread->floatTemp1;
+	U_32 index = *((U_32 *)currentThread->floatTemp2);
+	void *addr = NULL;
+	if (NULL == arrayObject) {
+		buildJITResolveFrameForRuntimeHelper(currentThread, parmCount);
+		addr = setCurrentExceptionFromJIT(currentThread, J9VMCONSTANTPOOL_JAVALANGNULLPOINTEREXCEPTION, NULL);
+	} else {
+		j9object_t value = NULL;
+		U_32 arrayLength = J9INDEXABLEOBJECT_SIZE(_currentThread, arrayObject);
+		if (index >= arrayLength) {
+			currentThread->tempSlot = (UDATA)index;
+			buildJITResolveFrameForRuntimeHelper(currentThread, parmCount);
+			addr = setCurrentExceptionFromJIT(currentThread, J9VMCONSTANTPOOL_JAVALANGARRAYINDEXOUTOFBOUNDSEXCEPTION, NULL);
+		} else {
+			buildJITResolveFrame(currentThread, J9_SSF_JIT_RESOLVE, parmCount);
+			PUSH_OBJECT_IN_SPECIAL_FRAME(currentThread, arrayObject);
+			value = currentThread->javaVM->internalVMFunctions->loadFlattenableArrayElement(currentThread, arrayObject, index, false);
```

Comment:
fast == false, so I need to build a special frame. Is this the correct way to do it in this context, or should I use OutOfLineNL?

Reply:
the frame should be built before call the function

Reply:
So I build it regardless of the path being fast or slow?